Apply For Loan option terminating with error

@rrpawar96  The error below is noticed and application terminates on the **Apply For Loan** function :

Selecting this option and waiting for a few seconds when the option screen is displayed results in the status in the image below:

![image](https://user-images.githubusercontent.com/97407144/149210555-2d99a654-2745-4669-8c4d-95f1dd29a0e6.png)


I/MultiDex: VM has multidex support, MultiDex support library is disabled.
W/art: Before Android 4.1, method android.graphics.PorterDuffColorFilter androidx.vectordrawable.graphics.drawable.VectorDrawableCompat.updateTintFilter(android.graphics.PorterDuffColorFilter, android.content.res.ColorStateList, android.graphics.PorterDuff$Mode) would have incorrectly overridden the package-private method in android.graphics.drawable.Drawable
D/AndroidRuntime: Shutting down VM
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.mifos.mobile, PID: 26240
    java.lang.RuntimeException: Unable to start activity ComponentInfo{org.mifos.mobile/org.mifos.mobile.ui.activities.HomeActivity}: android.os.BadParcelableException: Parcelable protocol requires a Parcelable.Creator object called CREATOR on class org.mifos.mobile.models.client.Client
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2416)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2476)
        at android.app.ActivityThread.-wrap11(ActivityThread.java)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1344)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:148)
        at android.app.ActivityThread.main(ActivityThread.java:5417)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
     Caused by: android.os.BadParcelableException: Parcelable protocol requires a Parcelable.Creator object called CREATOR on class org.mifos.mobile.models.client.Client
        at android.os.Parcel.readParcelableCreator(Parcel.java:2415)
        at android.os.Parcel.readParcelable(Parcel.java:2337)
        at android.os.Parcel.readValue(Parcel.java:2243)
        at android.os.Parcel.readArrayMapInternal(Parcel.java:2592)
        at android.os.BaseBundle.unparcel(BaseBundle.java:221)
        at android.os.BaseBundle.getInt(BaseBundle.java:795)
        at androidx.appcompat.app.AppCompatDelegateImpl.onCreate(AppCompatDelegateImpl.java:291)
        at androidx.appcompat.app.AppCompatActivity.onCreate(AppCompatActivity.java:73)
        at org.mifos.mobile.ui.activities.HomeActivity.onCreate(HomeActivity.kt:73)
        at android.app.Activity.performCreate(Activity.java:6237)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1107)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2369)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2476) 
        at android.app.ActivityThread.-wrap11(ActivityThread.java) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1344) 
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loop(Looper.java:148) 
        at android.app.ActivityThread.main(ActivityThread.java:5417) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616) 

--- a/app/src/main/java/org/mifos/mobile/models/client/Client.kt
+++ b/app/src/main/java/org/mifos/mobile/models/client/Client.kt
@@ -1,132 +1,60 @@
 package org.mifos.mobile.models.client
 
-import android.os.Parcel
 import android.os.Parcelable
+import kotlinx.android.parcel.Parcelize
 import org.mifos.mobile.models.Timeline
-import java.util.*
 
 /**
  * @author Vishwajeet
  * @since 20/06/16
  */
-class Client : Parcelable {
-    var id = 0
+@Parcelize
+data class Client(
+    var id: Int = 0,
 
-    var accountNo: String? = null
+    var accountNo: String? = null,
 
-    private var status: Status? = null
+    private var status: Status? = null,
 
-    private var active: Boolean? = null
+    private var active: Boolean? = null,
 
-    var activationDate: List<Int?>? = ArrayList()
+    var activationDate: List<Int> = ArrayList(),
 
-    var dobDate: List<Int?> = ArrayList()
+    var dobDate: List<Int> = ArrayList(),
 
-    var firstname: String? = null
+    var firstname: String? = null,
 
-    var middlename: String? = null
+    var middlename: String? = null,
 
-    var lastname: String? = null
+    var lastname: String? = null,
 
-    var displayName: String? = null
+    var displayName: String? = null,
 
-    var fullname: String? = null
+    var fullname: String? = null,
 
-    private var officeId: Int? = null
+    private var officeId: Int? = null,
 
-    var officeName: String? = null
+    var officeName: String? = null,
 
-    private var staffId: Int? = null
+    private var staffId: Int? = null,
 
-    private var staffName: String? = null
+    private var staffName: String? = null,
 
-    private var timeline: Timeline? = null
+    private var timeline: Timeline? = null,
 
-    var imageId = 0
+    var imageId: Int = 0,
 
-    var isImagePresent = false
+    var isImagePresent: Boolean = false,
 
-    private var externalId: String? = null
+    private var externalId: String? = null,
 
-    var mobileNo: String? = null
+    var mobileNo: String? = null,
 
-    var clientClassification: ClientClassification? = null
+    var clientClassification: ClientClassification? = null,
 
-    var clientType: ClientType? = null
+    var clientType: ClientType? = null,
 
-    var gender: Gender? = null
+    var gender: Gender? = null,
 
     var groups: List<Group> = ArrayList()
-    override fun toString(): String {
-        return "Client{" +
-                "id=" + id +
-                ", accountNo='" + accountNo + '\'' +
-                ", activationDate=" + activationDate +
-                ", firstname='" + firstname + '\'' +
-                ", middlename='" + middlename + '\'' +
-                ", lastname='" + lastname + '\'' +
-                ", displayName='" + displayName + '\'' +
-                ", fullname='" + fullname + '\'' +
-                '}'
-    }
-
-    override fun describeContents(): Int {
-        return 0
-    }
-
-    override fun writeToParcel(dest: Parcel, flags: Int) {
-        dest.writeInt(id)
-        dest.writeString(accountNo)
-        dest.writeParcelable(status, flags)
-        dest.writeValue(active)
-        dest.writeList(activationDate)
-        dest.writeList(dobDate)
-        dest.writeString(firstname)
-        dest.writeString(middlename)
-        dest.writeString(lastname)
-        dest.writeString(displayName)
-        dest.writeString(fullname)
-        dest.writeValue(officeId)
-        dest.writeString(officeName)
-        dest.writeValue(staffId)
-        dest.writeString(staffName)
-    }
-
-    constructor()
-    protected constructor(`in`: Parcel) {
-        id = `in`.readInt()
-        accountNo = `in`.readString()
-        status = `in`.readParcelable(Status::class.java.classLoader)
-        active = `in`.readValue(Boolean::class.java.classLoader) as Boolean
-        activationDate = ArrayList()
-        `in`.readList(activationDate, Int::class.java.classLoader)
-        dobDate = ArrayList()
-        `in`.readList(dobDate, Int::class.java.classLoader)
-        firstname = `in`.readString()
-        middlename = `in`.readString()
-        lastname = `in`.readString()
-        displayName = `in`.readString()
-        fullname = `in`.readString()
-        officeId = `in`.readValue(Int::class.java.classLoader) as Int
-        officeName = `in`.readString()
-        staffId = `in`.readValue(Int::class.java.classLoader) as Int
-        staffName = `in`.readString()
-        timeline = `in`.readParcelable(Timeline::class.java.classLoader)
-        fullname = `in`.readString()
-        imageId = `in`.readInt()
-        isImagePresent = `in`.readByte().toInt() != 0
-        externalId = `in`.readString()
-    }
-
-    companion object {
-        val CREATOR: Parcelable.Creator<Client?> = object : Parcelable.Creator<Client?> {
-            override fun createFromParcel(source: Parcel): Client? {
-                return Client(source)
-            }
-
-            override fun newArray(size: Int): Array<Client?> {
-                return arrayOfNulls(size)
-            }
-        }
-    }
-}
+): Parcelable

--- a/app/src/main/java/org/mifos/mobile/models/client/ClientClassification.kt
+++ b/app/src/main/java/org/mifos/mobile/models/client/ClientClassification.kt
@@ -1,12 +1,15 @@
 package org.mifos.mobile.models.client
 
+import android.os.Parcelable
+import kotlinx.android.parcel.Parcelize
+
 /**
  * Created by dilpreet on 10/7/17.
  */
-
+@Parcelize
 data class ClientClassification(
         var id: Int,
         var name: String? = null,
         var active: Boolean,
         var mandatory: Boolean
-)
+) : Parcelable

--- a/app/src/main/java/org/mifos/mobile/models/client/ClientType.kt
+++ b/app/src/main/java/org/mifos/mobile/models/client/ClientType.kt
@@ -1,12 +1,16 @@
 package org.mifos.mobile.models.client
 
+import android.os.Parcelable
+import kotlinx.android.parcel.Parcelize
+
 /**
  * Created by dilpreet on 10/7/17.
  */
 
+@Parcelize
 data class ClientType(
         var id: Int,
         var name: String? = null,
         var active: Boolean,
         var mandatory: Boolean
-)
+): Parcelable

--- a/app/src/main/java/org/mifos/mobile/models/client/Gender.kt
+++ b/app/src/main/java/org/mifos/mobile/models/client/Gender.kt
@@ -1,12 +1,15 @@
 package org.mifos.mobile.models.client
 
+import android.os.Parcelable
+import kotlinx.android.parcel.Parcelize
+
 /**
  * Created by dilpreet on 10/7/17.
  */
-
+@Parcelize
 data class Gender(
         var id: Int,
         var name: String? = null,
         var active: Boolean,
         var mandatory: Boolean
-)
+): Parcelable

--- a/app/src/main/java/org/mifos/mobile/models/client/Group.kt
+++ b/app/src/main/java/org/mifos/mobile/models/client/Group.kt
@@ -1,11 +1,14 @@
 package org.mifos.mobile.models.client
 
+import android.os.Parcelable
+import kotlinx.android.parcel.Parcelize
+
 /**
  * Created by dilpreet on 10/7/17.
  */
-
+@Parcelize
 data class Group(
         var id: Int,
         var accountNo: String? = null,
         var name: String? = null
-)
+): Parcelable

--- a/app/src/main/java/org/mifos/mobile/ui/fragments/HomeOldFragment.kt
+++ b/app/src/main/java/org/mifos/mobile/ui/fragments/HomeOldFragment.kt
@@ -165,7 +165,7 @@ class HomeOldFragment : BaseFragment(), HomeOldView, OnRefreshListener {
     override fun onActivityCreated(savedInstanceState: Bundle?) {
         super.onActivityCreated(savedInstanceState)
         if (savedInstanceState != null) {
-            showUserDetails(savedInstanceState.getParcelable<Parcelable>(Constants.USER_DETAILS) as Client)
+            showUserDetails(savedInstanceState.getParcelable<Parcelable>(Constants.USER_DETAILS) as? Client)
             presenter?.setUserProfile(preferencesHelper?.userProfileImage)
             showLoanAccountDetails(savedInstanceState.getDouble(Constants.TOTAL_LOAN))
             showSavingAccountDetails(savedInstanceState.getDouble(Constants.TOTAL_SAVINGS))

